build-logic/binary-compatibility/src/main/groovy/gradlebuild/binarycompatibility/upgrades/UpgradedProperties.java

```diff
@@ -0,0 +1,134 @@
+        return isOldMethod(jApiMethod, upgradedMethods, SETTER_REGEX);
+    }
+
+    private static boolean isOldGetterOfUpgradedProperty(JApiMethod jApiMethod, Map<String, UpgradedProperty> upgradedMethods) {
+        return isOldMethod(jApiMethod, upgradedMethods, GETTER_REGEX);
+    }
+
+    private static boolean isOldBooleanGetterOfUpgradedProperty(JApiMethod jApiMethod, Map<String, UpgradedProperty> upgradedMethods) {
+        return isOldMethod(jApiMethod, upgradedMethods, BOOLEAN_GETTER_REGEX) && jApiMethod.getReturnType().getOldReturnType().equals("boolean");
+    }
+
+    private static boolean isOldMethod(JApiMethod jApiMethod, Map<String, UpgradedProperty> upgradedMethods, Pattern pattern) {
+        String name = jApiMethod.getName();
+        if (!pattern.matcher(name).matches()) {
+            return false;
+        }
+        String descriptor = jApiMethod.getOldMethod().get().getSignature();
+        String containingType = jApiMethod.getjApiClass().getFullyQualifiedName();
+        return upgradedMethods.containsKey(getMethodKey(containingType, name, descriptor));
+    }
+
+    private static boolean isCurrentGetterOfUpgradedProperty(JApiMethod jApiMethod, Map<String, UpgradedProperty> currentMethods) {
+        return isCurrentMethod(jApiMethod, currentMethods, GETTER_REGEX);
+    }
+
+    private static boolean isCurrentMethod(JApiMethod jApiMethod, Map<String, UpgradedProperty> currentMethods, Pattern pattern) {
+        String name = jApiMethod.getName();
+        if (!pattern.matcher(name).matches()) {
+            return false;
+        }
+        String descriptor = jApiMethod.getNewMethod().get().getSignature();
+        String containingType = jApiMethod.getjApiClass().getFullyQualifiedName();
+        return currentMethods.containsKey(getMethodKey(containingType, name, descriptor));
+    }
+
+    public static Optional<String> maybeGetKeyOfOldMethodOfUpgradedProperty(JApiCompatibility jApiCompatibility, ViolationCheckContext context) {
+        if (!(jApiCompatibility instanceof JApiMethod) || !((JApiMethod) jApiCompatibility).getOldMethod().isPresent()) {
+            return Optional.empty();
+        }
+        JApiMethod jApiMethod = (JApiMethod) jApiCompatibility;
+        String name = jApiMethod.getName();
+        Map<String, UpgradedProperty> oldMethods = context.getUserData(OLD_METHODS_OF_UPGRADED_PROPERTIES);
+        String descriptor = jApiMethod.getOldMethod().get().getSignature();
+        String containingType = jApiMethod.getjApiClass().getFullyQualifiedName();
+        String key = getMethodKey(containingType, name, descriptor);
+        return oldMethods.containsKey(key) ? Optional.of(key) : Optional.empty();
+    }
+
+    public static String getMethodKey(String containingType, String methodName, String descriptor) {
+        return containingType + "#" + methodName + descriptor;
+    }
```

Comment:
IIUC, you started using `MethodKey` from `:model-core`. This is an internal API reference; if we change `MethodKey`, it will mess up the Gradleception build. I don't think it's a likely thing, but it would be good to avoid adding more internal references.

Also, we should at some point introduce checks to prevent more internal API references from our own plugins and from our build logic; that validation will flag this, too.

Reply:
This is not `MethodKey` from `:model-core` but it's my own implementation in build logic. I will rename the class, to avoid confusion.

Reply:
Fixed

Reply:
Ah, sorry, it's my mistake, should have checked out your branch.